python/lsst/consdb/hinfo.py

```diff
@@ -281,34 +281,29 @@ def altaz_midpoint_from_altaz(altaz_begin: AltAz, altaz_end: AltAz) -> AltAz:
     return AltAz(midpoint)
 
 
-def altaz_midpoint(
-    tracking_radec: SkyCoord,
-    datetime_begin: astropy.time.Time,
-    datetime_end: astropy.time.Time,
-    altaz_begin: AltAz,
-    altaz_end: AltAz,
-) -> AltAz:
+def altaz_midpoint(get_attribute: Callable[[str], Any]) -> AltAz:
```

Comment:
This seems kind of obscure. Why not just use `hasattr` to check if it has a specific attribute inside this function?